packages/vite/src/node/server/middlewares/static.ts

```diff
@@ -211,20 +212,29 @@ export function isFileServingAllowed(
   url: string,
   server: ViteDevServer,
 ): boolean {
-  if (!server.config.server.fs.strict) return true
+  const { config } = server
+  if (!config.server.fs.strict) return true
+  const filePath = fsPathFromUrl(url)
+  return isFileLoadingAllowed(config, filePath)
+}
 
-  const file = fsPathFromUrl(url)
+function isUriInFilePath(uri: string, filePath: string) {
+  return isSameFileUri(uri, filePath) || isParentDirectory(uri, filePath)
+}
 
-  if (server._fsDenyGlob(file)) return false
+export function isFileLoadingAllowed(
+  config: ResolvedConfig,
+  filePath: string,
+): boolean {
```

Comment:
I think aligning with existing signature is better than deprecating. There is really no reason to deprecate here

Reply:
The idea for the new signature is:
1. we use this in `transformRequest` where we no longer have access to the `server`. The server was not used at all, only the `config` is needed here.
2. we have the `filePath` when we use it, we don't need to call `fsPathFromUrl`

At least internally, we needed this new function. We don't need to expose this new signature in this PR though. And we certainly don't want to deprecate the old one at this point. We can leave the new one as internal, and then decide later on if we want to improve on this API (I think in general it is better to pass a config instead of the whole server if that's enough)

Reply:
I think it was exposed because it is needed in Vitest for example. I thought the problem was only with the order of arguments to be honest 😄

Reply:
Ah! Ya, now I remember you exposed it. About the order, I think is better to pass the config first (same if you need to pass a server or environment). If we later need to add an extra param, this form avoids ending up with config in the middle.

Reply:
I think having `config` first is nice too, but if we want to align it, we'd have to change `isFileServingAllowed` to have the correct position and probably refer to `config` only since it doesn't need a reference to the `server` now too.

Reply:
I was thinking about deprecating at one point `isFileServingAllowed` and only keep the new one later on, so we won't need to align then